.github/workflows/github-actions-tests.yml

```
@@ -15,27 +15,29 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12", "3.13" ]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13", "3.13t" ]
         mongodb-version: [ "4.4", "5.0", "6.0", "7.0", "8.0" ]
-        pydantic-version: [ "1.10.18", "2.9.2" , "2.10.4"]
+        pydantic-version: [ "1", "2"]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
+      - name: Install uv and set the python version
+        uses: astral-sh/setup-uv@v5
         with:
-          python-version: ${{ matrix.python-version }}
-          cache: pip
-          cache-dependency-path: pyproject.toml
+          enable-cache: true
+          cache-dependency-glob: "uv.lock"
       - name: Start MongoDB
         uses: supercharge/mongodb-github-action@1.11.0
         with:
           mongodb-version: ${{ matrix.mongodb-version }}
           mongodb-replica-set: test-rs
-      - name: install dependencies
-        run: pip install .[test,ci]
-      - name: install pydantic
-        run: pip install pydantic==${{ matrix.pydantic-version }}
+      - name: install dependencies for pydantic1
+        if: matrix.pydantic-version == '1'
+        run: uv sync --group test  --group ci --frozen && uv add "pydantic[email]==1.10.18" pydantic-extra-types
+      - name: install pydantic2 deps
+        if: matrix.pydantic-version == '2'
+        run: uv sync --group test  --group ci --frozen && uv add "pydantic[email]==2.11.3" pydantic-settings pydantic-extra-types
```

`uv sync --group test  --group ci --frozen` actually installs all other dependencies so the description/name "install dependencies for pydantic1" / "install pydantic2 deps" may be a bit misleading here...
Yes, dependencies for Pydantic v1 are being installed here along with all other test dependencies.
My suggestion is to perhaps split it into 3 steps. First install all "test" dependencies, then install either Pydantic v1 or Pydantic v2 dependencies.

```suggestion
      - name: Install test dependencies
        run: uv sync --group test --group ci --frozen
      - name: Install Pydantic v1 and its dependencies
        if: matrix.pydantic-version == '1'
        run: uv add "pydantic[email]==1.10.18" pydantic-extra-types
      - name: Install Pydantic v2 and its dependencies
        if: matrix.pydantic-version == '2'
        run: uv add "pydantic[email]==2.11.3" pydantic-settings pydantic-extra-types
```